src/test/test.cpp

```diff
@@ -600,69 +614,88 @@ recordOrCheckGlobalTestTxMetadata(TransactionMeta const& txMeta)
 }
 
 static void
-loadTestTxMeta(std::string const& path)
+loadTestTxMeta(stdfs::path const& dir)
 {
-    std::ifstream in(path);
-    if (!in)
+    if (!stdfs::is_directory(dir))
     {
-        logErrAndThrow(fmt::format("Failed to open {}", path));
+        logErrAndThrow(fmt::format("{} is not a directory", dir));
     }
-    in.exceptions(std::ios::failbit | std::ios::badbit);
-    Json::Value root;
-    in >> root;
     size_t n = 0;
-    for (auto entry = root.begin(); entry != root.end(); ++entry)
+    for (auto const& dirent : stdfs::directory_iterator{dir})
     {
-        std::string name = entry.key().asString();
-        std::vector<Hash> hashes;
-        for (auto const& h : *entry)
+        auto path = dirent.path();
```

Comment:
should filter out entries that are not files and files without the `json` extension?

Reply:
Fixed.